inc/plugins/hello.php

```diff
@@ -8,17 +8,58 @@
  *
  */
 
-// Disallow direct access to this file for security reasons
-if(!defined("IN_MYBB"))
+// Make sure we can't access this file directly from the browser.
+if(!defined('IN_MYBB'))
 {
-	die("Direct initialization of this file is not allowed.<br /><br />Please make sure IN_MYBB is defined.");
+	die('This file cannot be accessed directly.');
 }
+	
+// cache templates - this is important when it comes to performance
+// THIS_SCRIPT is defined by some of the MyBB scripts, including index.php
+if(defined('THIS_SCRIPT'))
+{
+    global $templatelist;
+
+    if(isset($templatelist))
+    {
+        $templatelist .= ',';
+    }
 
-$plugins->add_hook("pre_output_page", "hello_world");
-$plugins->add_hook("postbit", "hello_world_postbit");
+	if(THIS_SCRIPT== 'index.php')
```

Comment:
Why not

```php
if(THIS_SCRIPT == 'index.php' || THIS_SCRIPT == 'showthread.php')
{
	$templatelist .= 'hello_index, hello_message';
}
```

...or even `if in_array(THIS_SCRIPT, array('index.php', 'showthread.php'))`

Reply:
Just [without the space](https://github.com/mybb/mybb/pull/1389#r25568895) in the templatelist :wink:

Reply:
lol My comment was for line 28 👅

Reply:
Also, there needs to be a space before the `==` in both cases.